src/components/button/Button.tsx

```diff
@@ -1,35 +1,25 @@
-import { cssVar } from '../../models/cssType';
-import { ButtonM } from './Button.model';
-import styles from './Button.module.scss';
+import { Link } from "react-router-dom";
+import { type cssVar } from "../../models/cssType";
+import { type ButtonProps } from "./Button.model";
+import {
+  buttonPrimary,
+  buttonSecondary,
+} from "../../utils/styles/buttonStyles";
+import styles from "./Button.module.scss";
 
-const Button = ({
-     children,
-     bgcolor,
-     border,
-     color,
-     fontFamily,
-     fontSize
-}:ButtonM) => {
+const Button = ({ children, config }: ButtonProps) => {
+  const variantOptions: { [key: string]: cssVar } = {
+    buttonPrimary,
+    buttonSecondary,
+  };
 
-
-    const handleStylesButton: cssVar = {
-        "--bg-color": bgcolor,
-        "--border": border,
-        "--color": color,
-        "--font-family": fontFamily,
-        "--font-size": fontSize,
-    }
-
-    
+  const buttonStyles = variantOptions[config.variant];
 
   return (
-    <button 
-        className={ styles.container }
-        style={ handleStylesButton }    
-    >
-       { children }
-    </button>
-  )
-}
+    <Link to={config.route} className={styles.container} style={buttonStyles}>
```

Comment:
Why is the Button component rendering a Link? If its going to render a stylized Link the name of the component shouldn't be Button